README.markdown

```diff
@@ -1712,61 +2211,73 @@ Validates that all passed values are string data structures. Aborts catalog comp
 
 The following values pass:
 
-  ~~~
-  $my_string = "one two"
-  validate_string($my_string, 'three')
-  ~~~
+```puppet
+$my_string = "one two"
+validate_string($my_string, 'three')
+```
 
-  The following values fail, causing compilation to abort:
+The following values fail, causing compilation to terminate:
 
-  ~~~
-  validate_string(true)
-  validate_string([ 'some', 'array' ])
-  ~~~
+```puppet
+validate_string(true)
+validate_string([ 'some', 'array' ])
+```
 
-*Note:* validate_string(undef) will not fail in this version of the functions API (incl. current and future parser).
+*Note:* validate_string(`undef`) will not fail in this version of the functions API (including current and future parser). TODO: Is this true for Puppet 3 only?
```

Comment:
`undef` also validates as a string on my puppet 4.9, so I'd say on both 3 and 4.